transport/cluster.go

```diff
@@ -100,9 +172,18 @@ func (c *Cluster) refreshTopology() error {
 	}
 	// If node is present in both maps we can reuse its connection pool.
 	old := c.Peers()
-	m := PeerMap{}
-	for _, v := range rows {
-		n, err := cqlRowIntoNode(v)
+	peers := make(PeerMap)
+	dcRacks := make(DCRacksMap)
+	seenRacks := make(map[string]bool) // Key is rack name appended to its dc name.
+	all := make([]*Node, 0, len(old))
+	ring := *btree.New(BTreeDegree)
+	keyspaces, err := c.parseKeyspace()
```

Comment:
Maybe allocate topology struct pointer instead of doing

```
topology{
		peers:     peers,
		dcRacks:   dcRacks,
		allNodes:  all,
		ring:      ring,
		keyspaces: keyspaces,
	}
```
later?